### PR TITLE
Add 'import file' rule

### LINKED ISSUE #4: File name matching in default imports
```Js
import dataService from './data-service'
```
The variable and file name should match

Use this examples as a reference how to work with AST:

- https://medium.com/@btegelund/creating-an-eslint-plugin-87f1cb42767f

### PR DESCRIPTION
**Pull request by Artefall**
Resolves #4

### CODE REVIEW

File: test/import-file-and-var-name-matching.test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	},
+	{
+		code: `import SimpleComponent from './SimpleComponent.jsx';`
+	},
+	{
+		code: `import to_Complicated_Name from './to-complicated-name-service.js';`
+	},
+	{
+		code: `import to_Complicated_Name from './to-complicated-name-controller.js';`
+	}
```
Comment (constgen) [inline test/import-file-and-var-name-matching.test.js]: Add a case with importing from long paths
```js
import noop from '../../services/noop.js'
```

Comment (constgen) [inline test/import-file-and-var-name-matching.test.js]: Add a case with importing without an extension
```js
import noop from '../../services/noop.js'
```

File: test/import-file-and-var-name-matching.test.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		},
+		{
+			code: `import coffeInCafe from './to-complicated-name-service.js';`,
+			errors: [{ message: errorMessage }]
+		},
+		{
+			code: `import coffe_in_cafe from './to-complicated-name-controller.js';`,
+			errors: [{ message: errorMessage }]
+		}
+	]
```
Comment (constgen) [inline test/import-file-and-var-name-matching.test.js]: Add a case with importing from long paths

Comment (constgen) [reply test/import-file-and-var-name-matching.test.js]: Add a case with importing without an extension

File: test/import-file-and-var-name-matching.test.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+const imoprtFileAndVarNameMatching = require('../src/rules/import-file-and-var-name-matching');
+
+let RuleTester = require('eslint').RuleTester;
+
+let ruleTester = new RuleTester();	
+
+const errorMessage = `import file does not match with variable name`;
```
Comment (constgen) [inline test/import-file-and-var-name-matching.test.js]: **to match ~~with~~ something**

**Review by constgen — CHANGES_REQUESTED** [review]

File: test/import-file-and-var-name-matching.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Artefall) [reply test/import-file-and-var-name-matching.test.js]: Why should I need to ignore expression, when file name starts with numbers?

File: src/rules/import-file-and-var-name-matching/index.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
+let path = require('path');
+
+let flat = require('array-flatten');
+
+function removeFileExtension (string) {
+	let fileNameWIhoutExtension = path.parse(string).name;
```
Comment (constgen) [inline src/rules/import-file-and-var-name-matching/index.js]: `WIhout`

File: src/rules/import-file-and-var-name-matching/index.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+				let importVarName = node.specifiers[0].local.name;
+				let fileNameSource = node.source.value;
+				let varIsNotImported = !node.specifiers[0].imported;
+				let varNameEqualsToFileName = areFileNameAndVariableNameEqual(fileNameSource, importVarName);
+				let fileNameStartsWithNumber = isFileNameStartsWithNumber(fileNameSource);
+
+				if (!varNameEqualsToFileName && varIsNotImported && !fileNameStartsWithNumber) {
+					context.report({
+						node,
+						message: `import file does not match variable name`
```
Comment (constgen) [inline src/rules/import-file-and-var-name-matching/index.js]: We should ask user to fix the variable name and not the filename. Extra feature if we suggest what names are compatible with the given filename.

File: src/rules/import-file-and-var-name-matching/index.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+				let importVarName = node.declarations[0].id.name;
+				let fileNameSource = node.declarations[0].init.arguments[0].value;
+				let varNameEqualsToFileName = areFileNameAndVariableNameEqual(fileNameSource, importVarName);
+				let fileNameStartsWithNumber = isFileNameStartsWithNumber(fileNameSource);
+
+
+				if (!varNameEqualsToFileName && functionIsRequire && !fileNameStartsWithNumber) {
+					context.report({
+						node,
+						message: `import file does not match variable name`
```
Comment (constgen) [inline src/rules/import-file-and-var-name-matching/index.js]: We should ask user to fix the variable name and not the filename. Extra feature if we suggest what names are compatible with the given filename.

File: test/import-file-and-var-name-matching.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (constgen) [reply test/import-file-and-var-name-matching.test.js]: We have no solution for this. We are open for ideas

File: test/import-file-and-var-name-matching.test.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+	},
+	{
+		code: `import func from '10times.js'`
+	},
+	{
+		code: `import removeKebab from 'remove-kebab.js'`
+	}
+
+	
+	],
```
Comment (constgen) [inline test/import-file-and-var-name-matching.test.js]: We need a case for importing without a declaration. They should be ignored
```js
import './css/styles.css`
```

File: src/rules/import-file-and-var-name-matching/index.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+function splitKebabCase (stringToSplit) {
+	const KEBAB_CASE_SEPARATOR = '-';
+
+	return split(stringToSplit, KEBAB_CASE_SEPARATOR);
+}
+
+function isFileNameStartsWithNumber (pathToFile) {
+	let fileName = path.parse(pathToFile).name;
```
Comment (constgen) [inline src/rules/import-file-and-var-name-matching/index.js]: Duplicated code

**Review by constgen — CHANGES_REQUESTED** [review]
1. Add necessary test cases
2. Avoid hardcoded suffixes and determine them dinamically by this pattern `*.[suffix].{ext}`

**Review by constgen — CHANGES_REQUESTED** [review]
Not all comments are fixed

**Review by constgen — CHANGES_REQUESTED** [review]
Not all comments are fixed

**Review by constgen — APPROVED** [review]

**Pull request merged**